app/assets/javascripts/application.js

```diff
@@ -14,3 +14,106 @@
 //= require jquery_ujs
 
 //= require_tree .
+$(document).ready(function() {
+
+  var x = document.getElementById("demo");
+  var minUpdateUserDistance = 0.01 //km
+  var userId = $("div[style='display:none']").attr('id')
+  var firebaseRef = new Firebase("https://blinding-fire-43.firebaseio.com/");
```

Comment:
A couple things here:
- not sure why this is called `firebaseRef` - I assume you mean a reference to firebase?  It's not convention in Ruby to include the *type* of a variable in its name (although it is in other languages, such as C).
- you should keep your firebase url private.  Getting Ruby into JS is tricky business and there are a couple ways of doing it.  All the following ways assume you are storing the firebase url either a) in an environmental variable or b) in a YAML file that loads into a constant on initialization (like we discusssed friday).

#### Option 1: js erb file
Create a file in `app/views` with a `js.erb` extension.  Let's call it `app/views/firebase.js.erb`.
In that file you'd create a JavaScript variable and interpolate the url. 

```javascript
var FIREBASE_URL = <%= ENV["FIREBASE_URL"] %>;
```
You would then include that js file in your `layouts/application.html.erb`: `<%= javascript_include_tag '/firebase.js' %>`

#### Option 2: Write JS in your view
 You can use OOJS to create an object, which will hold the firebase url (and possibly other configuration options) in a normal JS file in `app/assets/javascripts`.  Let's say it's `app/assets/javascripts/firebase.js` 
```javascript
// create an empty object or use the existing variable
var AUDIOSCAPE = AUDIOSCAPE || {};
```  
Then in  your `layout/application.html.erb`, include some JS that sets a property on that variable with erb
```ruby
<script type='javascript'>
  AUDIOSCAPE.firebaseUrl = <%= FIREBASE_URL %>;
</script>
```